graylog2-server/src/main/java/org/graylog/events/rest/EventNotificationsResource.java

```diff
@@ -142,13 +152,43 @@ public Response update(@ApiParam(name = "notificationId") @PathParam("notificati
         }
 
         final ValidationResult validationResult = dto.validate();
+        validateEmailConfiguration(dto, validationResult);
         if (validationResult.failed()) {
             return Response.status(Response.Status.BAD_REQUEST).entity(validationResult).build();
         }
 
         return Response.ok().entity(resourceHandler.update(dto)).build();
     }
 
+    private void validateEmailConfiguration(NotificationDto dto, ValidationResult validationResult) {
+        if (dto.config() instanceof EmailEventNotificationConfig) {
+            EmailEventNotificationConfig emailEventNotificationConfig = (EmailEventNotificationConfig) dto.config();
+            if (!emailConfiguration.isEnabled()) {
+                validationResult.addError("config", "Email transport is not configured in graylog.conf");
+            }
+            if (isNullOrEmpty(emailConfiguration.getFromEmail()) && isNullOrEmpty(emailEventNotificationConfig.sender())) {
+                validationResult.addError("sender", "No default sender specified in graylog.conf. You must specify one here.");
+            } else {
+                if (!isNullOrEmpty(emailEventNotificationConfig.sender())) {
+                    try {
+                        InternetAddress email = new InternetAddress(emailEventNotificationConfig.sender());
+                        email.validate();
+                    } catch (AddressException e) {
+                        validationResult.addError("sender", "Invalid email address.");
+                    }
+                }
+                if (!isNullOrEmpty(emailConfiguration.getFromEmail())) {
```

Comment:
I wouldn't validate things from the `server.conf` here. The user might not be able to fix that himself.
We could add a validator to the `EmailConfiguration` class though.

Reply:
Or maybe it's okay to validate the serve.conf sender email here, but only if the one from the eventnotification is empty.

Reply:
That was also my first implementation. 
However, I found out that EmailFactory will always try to set the address from graylog.conf first if it's not empty. If it's not a valid address, sending fails with a AddressException. That's why I changed it to always validate.
The other option would be to change the logic of EmailSender altogether.

Reply:
hmm, okay. i think that warrants a validation then. Let's keep it as is :+1: